[YAML] aimodel_data_factory.py AssertionError: missing data col: binance:ETH/USDT:close

### To reproduce

I just fixed [#517](https://github.com/oceanprotocol/pdr-backend/issues/517) "aimodel_data_factory.py AssertionError: missing data col: binance:BTC/USDT:None" via [PR #518](https://github.com/oceanprotocol/pdr-backend/pull/518). 
- Root cause: ppss.yaml's aimodel_ss feeds section didn't have eg "c" or "ohlcv"; it assumed that they didn't need to be specified. This was an incorrect assumption: aimodel_ss needs it. In fact aimodel_ss class supports these signals, but the yaml file didn't have it.
- The fix was: _do_ specify signals in the ppss.yaml file. And add a test to aimodel_ss class constructor that complains if not specified
- From that fix, `pdr sim` would work, and unit tests would pass. PR got merged, ticket closed, good.

517 was testing on these input feeds: `binance BTC/USDT c`

I then tested on these: `binance BTC/USDT ohlcv'. It worked.

I then tested on these: `binance BTC/USDT ETH/USDT c`. It failed. Core traceback:
```text
  File "/Users/trentmc/code/pdr-backend/pdr_backend/aimodel/aimodel_data_factory.py", line 84, in create_xy
    assert hist_col in mergedohlcv_df.columns, f"missing data col: {hist_col}"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: missing data col: binance:ETH/USDT:close
```

Digging into the problem, I found the root cause: constructing of PPSS doesn't check for dependencies among feeds, and when some errors are caught inside other code, the error messages are opaque

give user-friendly error messages when the dependencies are violated. Rather, we sometimes get more opaque error messages for some violations (like we observed above).

To fix: 
- do check for dependencies among various ppss ss feeds
- if any of those checks fails, give a user-friendly error message
- have appropriate unit tests

### Full traceback

```text
(venv) trentmc@tlm-macbook: ~/code/pdr-backend $ pdr sim ppss.yaml 
dftool sim: Begin
Arguments:
PPSS_FILE=ppss.yaml
Start run
Get historical data, across many exchanges & pairs: begin.
  Data start: timestamp=1685577600000, dt=2023-06-01_00:00:00.000
  Data fin: timestamp=1704970520447, dt=2024-01-11_10:55:20.447
  Update all rawohlcv files: begin

    Update rawohlcv file at exchange=binance, pair=BTC/USDT: begin
      filename=/Users/trentmc/code/pdr-backend/parquet_data/binance_BTC-USDT_1h.parquet
      File already exists
      File starts at: timestamp=1685577600000, dt=2023-06-01_00:00:00.000
      File finishes at: timestamp=1704967200000, dt=2024-01-11_10:00:00.000
      User-specified start >= file start, so append file
      Aim to fetch data from start time: timestamp=1704970800000, dt=2024-01-11_11:00:00.000
      Given start time, no data to gather. Exit.

  Update all rawohlcv files: done
  Load rawohlcv file.
Get historical data, across many exchanges & pairs: done.
Traceback (most recent call last):
  File "/Users/trentmc/code/pdr-backend/./pdr", line 6, in <module>
    cli_module._do_main()
  File "/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trentmc/code/pdr-backend/pdr_backend/cli/cli_module.py", line 44, in _do_main
    func(args)
  File "/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trentmc/code/pdr-backend/pdr_backend/cli/cli_module.py", line 55, in do_sim
    sim_engine.run()
  File "/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trentmc/code/pdr-backend/pdr_backend/sim/sim_engine.py", line 92, in run
    self.run_one_iter(test_i, mergedohlcv_df)
  File "/Users/trentmc/code/pdr-backend/venv/lib/python3.11/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trentmc/code/pdr-backend/pdr_backend/sim/sim_engine.py", line 106, in run_one_iter
    X, y, _ = model_data_factory.create_xy(mergedohlcv_df, testshift)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trentmc/code/pdr-backend/pdr_backend/aimodel/aimodel_data_factory.py", line 84, in create_xy
    assert hist_col in mergedohlcv_df.columns, f"missing data col: {hist_col}"
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: missing data col: binance:ETH/USDT:close
```

--- a/pdr_backend/ppss/aimodel_ss.py
+++ b/pdr_backend/ppss/aimodel_ss.py
@@ -15,14 +15,13 @@ class AimodelSS(MultiFeedMixin, StrMixin):
     FEEDS_KEY = "input_feeds"
 
     def __init__(self, d: dict):
-        super().__init__(d)  # yaml_dict["aimodel_ss"]
+        super().__init__(
+            d, assert_feed_attributes=["signal"]
+        )  # yaml_dict["aimodel_ss"]
 
         # test inputs
         if self.approach not in APPROACHES:
             raise ValueError(self.approach)
-        for feed in self.feeds:
-            if feed.signal is None:
-                raise ValueError(f"Got signal=None for feed: {feed}")
 
         assert 0 < self.max_n_train
         assert 0 < self.autoregressive_n < np.inf

--- a/pdr_backend/ppss/base_ss.py
+++ b/pdr_backend/ppss/base_ss.py
@@ -18,9 +18,16 @@ class MultiFeedMixin:
         feeds = ArgFeeds.from_strs(self.feeds_strs)
 
         if assert_feed_attributes:
+            missing_attributes = []
             for attr in assert_feed_attributes:
                 for feed in feeds:
-                    assert getattr(feed, attr)
+                    if not getattr(feed, attr):
+                        missing_attributes.append(attr)
+
+            if missing_attributes:
+                raise AssertionError(
+                    f"Missing attributes {missing_attributes} for some feeds."
+                )
 
     # --------------------------------
     # yaml properties

--- a/pdr_backend/ppss/test/test_aimodel_ss.py
+++ b/pdr_backend/ppss/test/test_aimodel_ss.py
@@ -1,5 +1,6 @@
-import pytest
 import re
+
+import pytest
 from enforce_typing import enforce_types
 
 from pdr_backend.cli.arg_feed import ArgFeed

--- a/pdr_backend/ppss/test/test_aimodel_ss.py
+++ b/pdr_backend/ppss/test/test_aimodel_ss.py
@@ -1,4 +1,5 @@
 import pytest
+import re
 from enforce_typing import enforce_types
 
 from pdr_backend.cli.arg_feed import ArgFeed
@@ -72,5 +73,7 @@ def test_aimodel_ss_unhappy1():
     }
 
     # it should complain that it's missing a signal in input feeds
-    with pytest.raises(ValueError):
+    with pytest.raises(
+        AssertionError, match=re.escape("Missing attributes ['signal'] for some feeds")
+    ):
         AimodelSS(d)

